code/game/objects/items/miscellaneous.dm

```diff
@@ -381,3 +381,17 @@
 	name = "goat delivery beacon"
 	default_name = "Billy"
 	mob_choice = /mob/living/simple_animal/hostile/retaliate/goat
+
+/obj/item/choice_beacon/janicart
+	name = "janicart delivery beacon"
+	desc = "Summons a pod containing one (1) pimpin ride."
+
+/obj/item/choice_beacon/janicart/generate_display_names()
+	var/static/list/cart_spaghetti_list
+	if(!cart_spaghetti_list)
+		cart_spaghetti_list = list()
+		var/list/templist = list(/obj/vehicle/ridden/janicart/upgraded/keyless)
+		for(var/V in templist)
+			var/atom/A = V
+			cart_spaghetti_list[initial(A.name)] = A
+	return cart_spaghetti_list
```

Comment:
This doesn't need to be nearly this pointlessly complicated.

Reply:
The entire proc can be replaced with
```
return list("janitor cart" = /obj/vehicle/ridden/janicart/upgraded/keyless)
```